modules/logging/netwriter.go

```diff
@@ -196,7 +196,7 @@ func (reconn *redialerConn) Write(b []byte) (n int, err error) {
 		if err2 != nil {
 			// logger socket still offline; instead of discarding the log, dump it to stderr
 			os.Stderr.Write(b)
-			return
+			return 0, err2
```

Comment:
This changes the logic, but maybe in a good way. 

Might be worth considering returning `errors.Join` of both `err` and `err2` or maybe a new error wrapping both, tho.